src/ds3.c

```diff
@@ -1037,7 +1058,32 @@ void ds3_request_set_creation_date(ds3_request* _request, const char* creation_d
 
 void ds3_request_set_md5(ds3_request* _request, const char* md5) {
   struct _ds3_request* request = (struct _ds3_request*) _request;
-  request->md5 = ds3_str_init(md5);
+  request->checksum_type = MD5;
```

Comment:
It looks like these only have 2 spaces instead of 4.  I looks like you just continued what someone else has done, but we should correct this.